indicator_stochastic_rsi.go

```diff
@@ -0,0 +1,67 @@
+package techan
+
+import (
+	"math"
+
+	"github.com/sdcoffey/big"
+)
+
+type stochasticRSIIndicator struct {
+	curRSI Indicator
+	minRSI Indicator
+	maxRSI Indicator
+}
+
+// NewStochasticRSIIndicator returns a derivative Indicator which returns the stochastic RSI indicator for the given
+// RSI window.
+// https://www.investopedia.com/terms/s/stochrsi.asp
+func NewStochasticRSIIndicator(indicator Indicator, timeframe int) Indicator {
+	rsiIndicator := NewRelativeStrengthIndexIndicator(indicator, timeframe)
+	return stochasticRSIIndicator{
+		curRSI: rsiIndicator,
+		minRSI: NewMinimumValueIndicator(rsiIndicator, timeframe),
+		maxRSI: NewMaximumValueIndicator(rsiIndicator, timeframe),
+	}
+}
+
+func (sri stochasticRSIIndicator) Calculate(index int) big.Decimal {
+	curRSI := sri.curRSI.Calculate(index)
+	minRSI := sri.minRSI.Calculate(index)
+	maxRSI := sri.maxRSI.Calculate(index)
+
+	if minRSI.EQ(maxRSI) {
+		return big.NewDecimal(math.Inf(1))
```

Comment:
Why `Inf` here? Wouldn't 0 be a better dummy value?

Reply:
The reason it's `Inf` is simply because dividing by 0 tends to infinity. I am not sure what the correct number should be, but if you think 0 is better I can change it.